src/app/outbound/resolve.rs

```diff
@@ -0,0 +1,110 @@
+use super::endpoint;
+use crate::api_resolve::Metadata;
+use crate::app::dst::DstAddr;
+use crate::dns::Suffix;
+use linkerd2_error::{Error, Recover};
+use linkerd2_exp_backoff::{ExponentialBackoff, ExponentialBackoffStream};
+use linkerd2_proxy_core::{resolve, Resolve};
+use linkerd2_proxy_resolve::{map_endpoint, recover};
+use linkerd2_request_filter as request_filter;
+use std::sync::Arc;
+use tower_grpc as grpc;
+
+pub fn resolve<R>(
+    suffixes: Vec<Suffix>,
+    backoff: ExponentialBackoff,
+    resolve: R,
+) -> map_endpoint::Resolve<
```

Comment:
TIOLI: What do you think about hiding this huge type behind a newtype? That would make errors shorter (and, I wonder if it would shave a little type inference time off?).